sixtyfps_runtime/corelib/sharedvector.rs

```diff
@@ -251,6 +245,32 @@ impl<T: Clone> SharedVector<T> {
             }
         }
     }
+
+    fn shrink(&mut self, new_len: usize) {
+        if self.len() == new_len {
+            return;
+        }
+
+        debug_assert!(
+            unsafe { self.inner.as_ref().header.refcount.load(atomic::Ordering::Relaxed) } == 1
+        );
+        // Safety: caller (and above debug_assert) must ensure that the array is not shared.
```

Comment:
that should be an `assert`  or the function need to be marked `unsafe`

In fact, if it was public, i'd say that shrink should in fact detach if it is shared, but only clone new_len element to the final vector (and only allocates (new_len)